pipeline/submodules/select_direction.py

```diff
@@ -114,6 +114,137 @@ def filter_fn(refusal_score, steering_score, kl_div_score, layer, n_layer, kl_th
         return True
     return False
 
+def select_direction_2(
+    model_base: ModelBase,
+    harmful_instructions,
+    harmless_instructions,
+    candidate_directions: Float[Tensor, ' d_model'],
+    artifact_dir,
+    kl_threshold=0.1, # directions larger KL score are filtered out
+    induce_refusal_threshold=0.0, # directions with a lower inducing refusal score are filtered out
+    prune_layer_percentage=0.2, # discard the directions extracted from the last 20% of the model
+    batch_size=32
+):
+    if not os.path.exists(artifact_dir):
+        os.makedirs(artifact_dir)
+
+    d_model = candidate_directions.shape
+    n_layer = 18
+        
+    direction_to_add = candidate_directions
+
+    baseline_refusal_scores_harmful = get_refusal_scores(model_base.model, harmful_instructions, model_base.tokenize_instructions_fn, model_base.refusal_toks, fwd_hooks=[], batch_size=batch_size)
+    baseline_refusal_scores_harmless = get_refusal_scores(model_base.model, harmless_instructions, model_base.tokenize_instructions_fn, model_base.refusal_toks, fwd_hooks=[], batch_size=batch_size)
```

Comment:
Can we rename refusal_scores to be consistent with the paper (afair, it is calculating the refusal_metric and not the score?) It's quite confusing :(